### DISCUSSION TITLE
Are the new switches with comfort sensors supported ?

### DISCUSSION THREAD

**Discussion by brett1996** [joleys/niko-home-control-II#171] [Q&A]
I'm planning the replace my digital black screen with the new Push button with LED and comfort sensors.

Are these already supported?

Article number: 100-52201

[https://www.niko.eu/en/products/niko-home-control/products-on-bus-wiring/single-push-button-with-led-and-comfort-sensors-productmodel-niko-d3fe0a78-5dbf-44cb-b59a-00cfa34a74de](url)

**Comment by tijsverkoyen** [comment]
To be honest: I don't know.

The API documentation does not state the article numbers. To mu knowledge buttons are not exposed through the API, only the "actions".

**Reply by brett1996** [reply]
After I posted this, I found out that there is something called Thermo switch, And I think that this is the same as the new product. And this has been in the API for almost a year, before releasing the product. 

I ordered a switch, and will try it.

**Reply by brett1996** [reply]
I have been through your code and found out that there is nothing programmed for this.
Is it an option that you can help me with this ?

**Reply by tijsverkoyen** [reply]
If you have the device installed and configured in Niko Home Control, you can [send me your device list](https://github.com/joleys/niko-home-control-II?tab=readme-ov-file#i-need-to-see-the-device-list).

With that I can add support, if Niko exposes the device through the API.

**Comment by brett1996** [comment]
Hi,

see the attachment.
Thanks for helping.

[home-assistant.log](https://github.com/user-attachments/files/16926681/home-assistant.log)

**Comment by tijsverkoyen** [comment]
@brett1996 How is the device (the Push Button with Led and comfort sensors) called in your Niko Home Control app or in the software.
Because I don't find anything in your device list that looks like `Thermo switch`. I see a lot of virtual thermostats but these are already supported and exposed

**Comment by brett1996** [comment]
was this helpful?

**Reply by tijsverkoyen** [reply]
See my last question: 

>How is the device (the Push Button with Led and comfort sensors) called in your Niko Home Control app or in the software.
Because I don't find anything in your device list that looks like Thermo switch. I see a lot of virtual thermostats but these are already supported and exposed

**Comment by brett1996** [comment]
hi,

i have sent you a screenshot of the niko page where the sensors can be
found.
They have the same name in the software.

[image: image.png]

**Comment by tijsverkoyen** [comment]
I did not receive an image. Probably because you are replying by email instead of using the GitHub interface: https://github.com/joleys/niko-home-control-II/discussions/171#discussioncomment-10809349

**Comment by brett1996** [comment]
![image](https://github.com/user-attachments/assets/a32ac366-6bbf-4d2a-8d4f-8eead31e12d3)

**Reply by tijsverkoyen** [reply]
These are exposed as thermostats in the Niko API. 

```json
        {
          "Uuid": "XXX",
          "Identifier": "XXX",
          "Model": "virtual",
          "Online": "True",
          "Name": "Schakelaar slaapkamer 2",
          "Technology": "nikohomecontrol",
          "Type": "thermostat",
          "Traits": [],
          "Parameters": [
            ...
          ],
          "Properties": [
            {
              "OverruleActive": "False"
            },
            {
              "SetpointTemperature": "7.00"
            },
            {
              "Demand": "None"
            },
            {
              "AmbientTemperature": "23.50"
            },
            {
              "OverruleSetpoint": ""
            },
            {
              "Program": "Off"
            },
            {
              "OverruleTime": ""
            },
            {
              "EcoSave": "False"
            }
          ],
          "PropertyDefinitions": [
            ...
          ]
        },
```

The thermostats are also exposed in Home Assistant. See the attached image.

![brett01](https://github.com/user-attachments/assets/ffee00f6-0fbe-43a3-b76b-bc2261254263)

**Reply by brett1996** [reply]
Hi, yes I know, I had Digital blacks before. But I replaced them with this product because these have Temperature & Humidity. And I thought that in die API documentation the Humidity was supported. But I don't believe it is already supported in this add-on. I will look up the specific chapter in the documentation tomorrow.

**Reply by tijsverkoyen** [reply]
@brett1996 the humidity is not exposed. The exposed Properties are:

```json
{
          "Uuid": "XXX",
          "Identifier": "XXX",
          "Model": "virtual",
          "Online": "True",
          "Name": "Schakelaar slaapkamer 2",
          "Technology": "nikohomecontrol",
          "Type": "thermostat",
          "Traits": [],
          "Parameters": [
            ...
          ],
          "Properties": [
            {
              "OverruleActive": "False"
            },
            {
              "SetpointTemperature": "7.00"
            },
            {
              "Demand": "None"
            },
            {
              "AmbientTemperature": "23.50"
            },
            {
              "OverruleSetpoint": ""
            },
            {
              "Program": "Off"
            },
            {
              "OverruleTime": ""
            },
            {
              "EcoSave": "False"
            }
          ],
          "PropertyDefinitions": [
            ...
          ]
        },
```

Humidity is not present. You could try to contact Niko Support and ask them to add it in a future version.
